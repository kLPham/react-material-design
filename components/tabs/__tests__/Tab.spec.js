import React from 'react';
import { shallow } from 'enzyme';
import Tab from '../tab';

describe('<Tab />', () => {
    it('should render', () => {
        const label = 'Home';
        const tab = shallow(
            <Tab label={label} />,
        );
        expect(tab).toMatchSnapshot();
    });

    it('should render active', () => {
        const label = 'Home';
        const active = true;
        const tab = shallow(
            <Tab active={active} label={label} />,
        );
        expect(tab).toMatchSnapshot();
    });

    it('should render with icon', () => {
        const icon = 'home';
        const tab = shallow(
            <Tab icon={icon} />,
        );
        expect(tab).toMatchSnapshot();
    });

    it('should render with icon and Text', () => {
        const icon = 'home';
        const label = 'Home';
        const tab = shallow(
            <Tab label={label} icon={icon} />,
        );
        expect(tab).toMatchSnapshot();
    });
});
