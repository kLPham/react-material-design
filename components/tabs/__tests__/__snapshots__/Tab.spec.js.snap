// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> should render 1`] = `
<a
  className="mdc-tab"
>
  Home
</a>
`;

exports[`<Tab /> should render active 1`] = `
<a
  className="mdc-tab mdc-tab--active"
>
  Home
</a>
`;

exports[`<Tab /> should render with icon 1`] = `
<a
  className="mdc-tab"
>
  <i
    aria-label="home"
    className="material-icons mdc-tab__icon"
  >
    home
  </i>
</a>
`;

exports[`<Tab /> should render with icon and Text 1`] = `
<a
  className="mdc-tab mdc-tab--with-icon-and-text"
>
  <i
    aria-label="home"
    className="material-icons mdc-tab__icon"
  >
    home
  </i>
  <span
    className="mdc-tab__icon-text"
  >
    Home
  </span>
</a>
`;
